Edit docs to clarify role of static folder

codesections opened the pull request:
The existing docs briefly mentions the static folder under `directory structure`, by stating that the static folder can "contain[] any kind of files. All the files/folders in the static folder will be copied as-is in the output directory."

However, the role of the static folder is not mentioned when discussing how to add content to the site.  This commit adds a short section to the `Content -- overview` section explaining how to use the static folder to add static content and providing a few use cases.

(I'm suggesting this addition because I spent a while earlier today struggling to add the keybase.txt file required for [my keybase proof](https://keybase.io/codesections).  I was looking for a way to have a file in `content` that wasn't parsed as markdown, and didn't realize that I should be saving the file to `static` and skipping `content` entierly.)

codesections committed:
Edit docs to clarify role of static folder

The existing docs briefly mentions the static folder under `directory
structure`, by stating that the static folder can "contain[] any kind of
files. All the files/folders in the static folder will be copied as-is in
the output directory."

However, the role of the static folder is not mentioned when discussing
how to add content to the site.  This commit adds a short section to the
`Content -- overview` section explaining how to use the static folder to
add static content and providing a few use cases.

Keats commented on docs/content/documentation/content/overview.md:
```diff
@@ -63,3 +63,14 @@ the public web site. You can achieve this by simply setting `ignored_content` in
+Typically, you might put site-wide assets (such as the site favicon, site logos or site-wide
+JavaScript) in the root of the static directory.  You can also place any HTML or other files that
+you wish to be included without modification (that is, without being parsed as Markdown files)
+into the static directory.
```
The last sentence is not 100% true, you can also put non-md files in `content` as colocated-assets (https://www.getgutenberg.io/documentation/content/overview/#assets-colocation) as mentioned above in that file and they won't be modified either.

codesections replied:
I added an example that clarifies that you can either colocate files or add them to the static folder, depending on which system fits your desired organazational structure.  Please let me know if you would like me to revise further.

codesections committed:
Merge remote-tracking branch 'upstream/master'

codesections committed:
Revise changes to documentation re: static folder

codesections committed:
Add `index page` section to documentation

The current documentation does not describe how to create a index page.
I initially found this confusing, because I expected an index page to be
a **page** rather than a section.  Thus, I tried to access the page
content with `{{ page.content }}` and was very frustrated when I could
not.

This addition clarifies that the index page is **always** a section,
even if it does not have any sub-pages.  This should also help people
who intend to use Gutenberg to build stand-alone webpages, rather than
blogs.

codesections committed:
Revert "Revise changes to documentation re: static folder"

This reverts commit 1c735c5bcb627e74a113395ed1afd553bb64dc9b.

codesections committed:
Revert "Revert "Revise changes to documentation re: static folder""

This reverts commit 5d6e04073416d869211a0cca5533ba25b973fea3.

codesections committed:
Revert "Add `index page` section to documentation"

This reverts commit 5e9549e143b79ec5b8f7b08f9329ede6fe807cb3.
